apidoc/Titanium/UI/WebView.yml

```diff
@@ -244,20 +244,114 @@ methods:
   - name: setBasicAuthentication
     summary: |
         Sets the basic authentication for this web view to use on subsequent URl requests.
+    description: |
+        Persistence property used in this method is only supported on iOS platform and 
+        from SDK version 7.4.0.
     platforms: [android, iphone, ipad]
+
     parameters:
 
       - name: username
-        summary: Basic auth username.
+        summary: Basic auth user name.
         type: String
 
       - name: password
         summary: Basic auth password.
         type: String
 
+      - name: persistence
+        summary: Constants specify how long the credential will be kept.
+        type: Number
+        constants: Titanium.UI.CREDENTIAL_PERSISTENCE_*
+
   - name: stopLoading
     summary: Stops loading a currently loading page.
 
+  - name: startListeningToProperties
+    summary: Add native properties for observing for change. 
+    description: |
+        Some common properties are title, URL, estimatedProgress etc. 
+        See native KVO capability at https://developer.apple.com/documentation/webkit/wkwebview.
+        See example section "Listening to Web View properties in iOS" for usage.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
+    parameters:
+      - name: propertyList
+        summary: List of properties for listen.
+        type: Array
+
+  - name: stopListeningToProperties
+    summary: Remove native properties from observing.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
+    parameters:
+      - name: propertyList
+        summary: List of properties to remove.
+        type: Array
+
+  - name: takeSnapshot
+    summary: Takes a snapshot of the view's visible viewport.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
+    osver: {ios: {min: "11.0"}}
+    parameters:
+      - name: callback
+        summary: Function to call upon snapshot captured.
+        type: Callback<SnapshotResult>
+
+  - name: addUserScript
+    summary: Adds a user script.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
+    parameters:
+      - name: source
+        summary: A script that can be injected into a webpage.
+        type: String
+      - name: injectionTime
+        summary: The time when the script should be injected into the webpage.
+        type: Number
+        constants: Titanium.UI.INJECTION_TIME_DOCUMENT_*
+      - name: mainFrameOnly
+        summary: |
+            A Boolean value indicating whether the script should be injected only into the 
+            main frame (true) or into all frames (false).
+        type: Boolean
+
+  - name: removeAllUserScripts
+    summary: Removes all associated user scripts.      
+    platforms: [iphone, ipad]
+    since: "7.4.0"
```

Comment:
How will `addUserScript()` and `removeAllUserScripts()` impact our existing `evalJS()` function and our support for Titanium JS APIs within a locally loaded HTML file? Both of those things involve script injection as well. Also, will the `addUserScript()` and `removeAllUserScripts()` method only work **after** the "load" event has been received? If so, then we should document this... or have the given scripts queued to be added once loaded.

One more thing. Please run the test in PR #9931 . This loads local HTML in string form and verifies that it has access to our Titanium APIs... and it supports 2 way interop between the WebView and the Titanium "app.js".